src/code/z_kankyo.c

```diff
@@ -1,94 +1,2494 @@
+        { 0x5556, 0xAAAB, 1, 1 },
+        { 0xAAAB, 0xB556, 1, 2 },
+        { 0xB556, 0xCAAC, 2, 3 },
+        { 0xCAAC, 0xFFFF, 3, 3 },
+    },
+    {
+        { 0x0000, 0x2AAC, 7, 7 },
+        { 0x2AAC, 0x4000, 7, 4 },
+        { 0x4000, 0x5556, 4, 5 },
+        { 0x5556, 0xAAAB, 5, 5 },
+        { 0xAAAB, 0xB556, 5, 6 },
+        { 0xB556, 0xCAAC, 6, 7 },
+        { 0xCAAC, 0xFFFF, 7, 7 },
+    },
+    {
+        { 0x0000, 0x2AAC, 11, 11 },
+        { 0x2AAC, 0x4000, 11, 8 },
+        { 0x4000, 0x5556, 8, 9 },
+        { 0x5556, 0xAAAB, 9, 9 },
+        { 0xAAAB, 0xB556, 9, 10 },
+        { 0xB556, 0xCAAC, 10, 11 },
+        { 0xCAAC, 0xFFFF, 11, 11 },
+    },
+    {
+        { 0x0000, 0x2AAC, 15, 15 },
+        { 0x2AAC, 0x4000, 15, 12 },
+        { 0x4000, 0x5556, 12, 13 },
+        { 0x5556, 0xAAAB, 13, 13 },
+        { 0xAAAB, 0xB556, 13, 14 },
+        { 0xB556, 0xCAAC, 14, 15 },
+        { 0xCAAC, 0xFFFF, 15, 15 },
+    },
+    {
+        { 0x0000, 0x2AAC, 23, 23 },
+        { 0x2AAC, 0x4000, 23, 20 },
+        { 0x4000, 0x5556, 20, 21 },
+        { 0x5556, 0xAAAB, 21, 21 },
+        { 0xAAAB, 0xB556, 21, 22 },
+        { 0xB556, 0xCAAC, 22, 23 },
+        { 0xCAAC, 0xFFFF, 23, 23 },
+    },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_80073988.s")
+struct_8011FC1C D_8011FC1C[][9] = {
+    {
+        { 0x0000, 0x2AAC, 0, 3, 3 },
+        { 0x2AAC, 0x3556, 1, 3, 0 },
+        { 0x3556, 0x4000, 0, 0, 0 },
+        { 0x4000, 0x5556, 1, 0, 1 },
+        { 0x5556, 0xAAAB, 0, 1, 1 },
+        { 0xAAAB, 0xB556, 1, 1, 2 },
+        { 0xB556, 0xC001, 0, 2, 2 },
+        { 0xC001, 0xCAAC, 1, 2, 3 },
+        { 0xCAAC, 0xFFFF, 0, 3, 3 },
+    },
+    {
+        { 0x0000, 0x2AAC, 0, 7, 7 },
+        { 0x2AAC, 0x3556, 1, 7, 4 },
+        { 0x3556, 0x4000, 0, 4, 4 },
+        { 0x4000, 0x5556, 1, 4, 5 },
+        { 0x5556, 0xAAAB, 0, 5, 5 },
+        { 0xAAAB, 0xB556, 1, 5, 6 },
+        { 0xB556, 0xC001, 0, 6, 6 },
+        { 0xC001, 0xCAAC, 1, 6, 7 },
+        { 0xCAAC, 0xFFFF, 0, 7, 7 },
+    },
+    {
+        { 0x0000, 0x1556, 0, 3, 3 },
+        { 0x1556, 0x2AAC, 1, 3, 0 },
+        { 0x2AAC, 0x5556, 0, 0, 0 },
+        { 0x5556, 0x6AAB, 1, 0, 1 },
+        { 0x6AAB, 0x9556, 0, 1, 1 },
+        { 0x9556, 0xAAAB, 1, 1, 2 },
+        { 0xAAAB, 0xD556, 0, 2, 2 },
+        { 0xD556, 0xEAAB, 1, 2, 3 },
+        { 0xEAAB, 0xFFFF, 0, 3, 3 },
+    },
+    {
+        { 0x0000, 0x3556, 0, 11, 11 },
+        { 0x3556, 0x4000, 1, 11, 8 },
+        { 0x4000, 0x4AAB, 0, 8, 8 },
+        { 0x4AAB, 0x5556, 1, 8, 9 },
+        { 0x5556, 0xAAAB, 0, 9, 9 },
+        { 0xAAAB, 0xB556, 1, 9, 10 },
+        { 0xB556, 0xC001, 0, 10, 10 },
+        { 0xC001, 0xCAAC, 1, 10, 11 },
+        { 0xCAAC, 0xFFFF, 0, 11, 11 },
+    },
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kankyo/func_80073A5C.s")
+SkyboxFile gSkyboxFiles[] = {
+    {
+        { _vr_fine0_staticSegmentRomStart, _vr_fine0_staticSegmentRomEnd },
```

Comment:
We should probably reuse the `ROM_FILE` macro defined in `z_scene.c` here. We could move the few `ROM_FILE*` macros to `macros.h` or `z64.h` for now so they are usable everywhere.